app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,355 @@
+            mNoGroupCardsText.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu inputMenu)
+    {
+        this.mMenu = inputMenu;
+
+        getMenuInflater().inflate(R.menu.manage_group_menu, inputMenu);
+
+        MenuItem confirmButton = inputMenu.findItem(R.id.action_confirm);
+        Drawable icon = confirmButton.getIcon();
+        icon.mutate();
+        icon.setTint(Color.WHITE);
+        confirmButton.setIcon(icon);
+        return super.onCreateOptionsMenu(inputMenu);
+    }
+
+    private void leaveWithoutSaving(){
+        if (hasChanged()){
+            AlertDialog.Builder builder = new AlertDialog.Builder(ManageGroupActivity.this);
+            builder.setTitle(R.string.discard_changes);
+            builder.setMessage(R.string.discard_changes_confirm);
+            builder.setPositiveButton(R.string.confirm, (dialog, which) -> finish());
+            builder.setNegativeButton(R.string.cancel, (dialog, which) -> dialog.dismiss());
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        }else{
+            finish();
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem inputItem)
+    {
+        int id = inputItem.getItemId();
+
+        if (id == R.id.action_confirm)
+        {
+            String currentGroupName = mGroupNameText.getText().toString();
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                if(!mGroupNameNotInUse) {
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_already_in_use, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return true;
+                }
+                if(currentGroupName.trim().length() == 0){
+                    Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_is_empty, Toast.LENGTH_SHORT);
+                    toast.show();
+                    return true;
+                }
+            }
+
+            mAdapter.commitToDatabase(getApplicationContext(), mGroup._id);
+            Toast toast = Toast.makeText(getApplicationContext(), R.string.group_updated, Toast.LENGTH_SHORT);
+            if(!currentGroupName.trim().equals(mGroup._id)){
+                mDB.updateGroup(mGroup._id, currentGroupName.trim());
+            }
+            toast.show();
+            finish();
+            return true;
+        }
+
+        return super.onOptionsItemSelected(inputItem);
+    }
+    @Override
+    public void onBackPressed()
+    {
+        leaveWithoutSaving();
+    }
+    
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
+        return true;
+    }
+
+    private boolean hasChanged(){
+        return mAdapter.hasChanged() || !mGroup._id.equals(mGroupNameText.getText().toString().trim());
+    }
+
+    private void setSort(DBHelper.LoyaltyCardOrder order, DBHelper.LoyaltyCardOrderDirection direction) {
```

Comment:
I honestly don't think it is worth implementing sorting and all in here, way too much complexity.